### PR TITLE
Add MultiStateAppDeployer interface

### PR DESCRIPTION
**Pull request by markpollack**

### CODE REVIEW

File: spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package org.springframework.cloud.deployer.spi.app;
+
+import java.util.Map;
+
+/**
+ * Extension of the AppDeployer interface that adds an additional
+ * method to return the DeploymentState for a collection of deployment ids.
+ */
+public interface MultiStateAppDeployer extends AppDeployer {
```
Comment (markfisher) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java]: should go ahead and add `@Deprecated` with a mention that the method will move in a future release (likely 2.0)

Comment (markpollack) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java]: will do!

File: spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java [hunk 2]
```diff
@@ -20,7 +20,14 @@
 /**
  * Extension of the AppDeployer interface that adds an additional
  * method to return the DeploymentState for a collection of deployment ids.
+ *
+ * This method has been introduced for a tactical reason to provide an optimization
```
Comment (trisberg) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java]: should we add JavaDoc `@deprecated`?

Comment (markfisher) [inline spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/MultiStateAppDeployer.java]: yes

**Comment by trisberg** [comment]
Merged with @deprecated added to JavaDoc as commit 43220ac4dd94b4fc875db920471d55f91e765984

**Pull request closed without merging**